app/utils/wiki-handlers/article.js

```diff
@@ -34,42 +34,63 @@ function afterModel(route, model) {
 }
 
 /**
- * This function tracks page view only on articles on Lyrics Wiki (id: 43339).
+ * This function, along with shouldSendLyricFindRequest, tracks page view only on articles on Lyrics Wiki (id: 43339).
  * Notice that params amgid and gracenoteid are set to 0, those params are not important,
  * but to be consistent with Oasis we send them
  *
  * https://github.com/Wikia/app/blob/dev/extensions/3rdparty/LyricWiki/LyricFind/js/modules/LyricFind.Tracker.js
  *
- * @param {Ember.model} model
- * @param {number} wikiId
+ * @param {Ember.Object} model
  * @param {String} host
+ * @param {Ember.Object} logger - logger service
+ * @param {Ember.Object} headers - FastBoot request's headers
  */
-function sendLyricsPageView(model, wikiId, host) {
-	if (wikiId === 43339 && !model.get('isMainPage')) {
-		fetch(buildUrl({
-			host,
-			path: '/wikia.php',
-			query: {
-				controller: 'LyricFind',
-				method: 'track',
-				title: model.get('title'),
-				amgid: 0,
-				gracenoteid: 0,
-				rand: (`${Math.random()}`).substr(2, 8)
-			}
-		}));
-	}
+function sendLyricsPageView({model, host, logger, headers}) {
+	fetch(buildUrl({
+		host,
+		path: '/wikia.php',
+		query: {
+			controller: 'LyricFind',
+			method: 'track',
+			title: model.get('title'),
+			amgid: 0,
+			gracenoteid: 0,
+			rand: (`${Math.random()}`).substr(2, 8)
+		}
+	})).then(() => {
+		logger.info('LyricFind PageView tracking event sent', {headers: headers.getAll()});
+	});
+}
+
+/**
+ * @param {Ember.Object} model
+ * @param {number} wikiId
+ * @param {Ember.Object} headers - FastBoot request's headers
+ * @returns {boolean}
+ */
+function shouldSendLyricFindRequest({model, wikiId, headers}) {
+	const lyricWikiId = 43339;
+
+	// 'goreplay' header indicates http traffic replayed between SJC and RES
+	// for more info please read 'Confluence > Operations > Goreplay' article
+	return wikiId === lyricWikiId
+		&& !model.get('isMainPage')
+		&& headers.get('X-Wikia-Is-Internal-Request') !== 'goreplay';
```

Comment:
Honestly, the presence of the `X-Wikia-Is-Internal-Request` header (regardless of value) is probably reason enough not to send a beacon.  But this stop the bulk of the problem for now anyway.